src/commands/DensityProfile.hpp

```diff
@@ -17,29 +17,38 @@ class DensityProfile final: public AveCommand {
 	std::string outfile;
 	/// Selection for the donor-acceptor
 	std::string selection;
-        /// Type of profile
-        std::string type_profile;
-        /// Coordinate of origin
+	/// Coordinate of origin
         Vector3D origin;
         /// Number of points in the profile
-        size_t npoints;
+        size_t npoints[2];
         /// Maximum in the profile
-        double max = 0;
+        double max[2] = {0, 0};
         /// Minimum in the profile
-        double min = 0;
+        double min[2] = {0, 0};
     };
 
-    DensityProfile(): selection_("atoms: all"), axis_(0,0,1) {}
+    DensityProfile(): selection_("atoms: all"), axis_() {}
     std::string description() const override;
 
     Averager<double> setup(int argc, const char* argv[]) override;
     void accumulate(const chemfiles::Frame& frame, Histogram<double>& histogram);
     void finish(const Histogram<double>& histogram);
 
+    void insert_axis(Axis axis) { axis_.push_back(axis);}
+    size_t dimensionality() { return axis_.size();}
+    const Axis& get_axis(size_t dimension) const {
```

Comment:
This function is not needed either =). You can use `std::vector::at` if you want bounds checking.